### ISSUE TITLE
wasm wrangling!

### ISSUE THREAD

**Issue by crisdosaygo** [DOSAYGO-STUDIO/rain#44]
From a recent commit (7ec80ce0584872a411ebbe77ec854b4afbd66060):

```
wasm works except for 1 hash string
also the hashes are different in the browser versus node for the same wasm module...

Current JS hash vectors with node:

e3ea5f8885f7bb16468d08c578f0e7cc15febd31c27e323a79ef87c35756ce1e ""
9e07ce365903116b62ac3ac0a033167853853074313f443d5b372f0225eede50 "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
f88600f4b65211a95c6817d0840e0fc2d422883ddf310f29fa8d4cbfda962626 "The quick brown fox jumps over the lazy dog"
ec05208dd1fbf47b9539a761af723612eaa810762ab7a77b715fcfb3bf44f04a "The quick brown fox jumps over the lazy cog"
974bfe75e0f1b1b4d95bba5e11577f2b29dd3c27b0ed6645effea070c25fde71 "The quick brown fox jumps over the lazy dog." MISMATCH!
410427b981efa6ef884cd1f3d812c880bc7a37abc7450dd62803a4098f28d0f1 "After the rainstorm comes the rainbow."
47b5d8cb1df8d81ed23689936d2edaa7bd5c48f5bc463600a4d7a56342ac80b9 "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"

the 'correct' C++ vectors:

e3ea5f8885f7bb16468d08c578f0e7cc15febd31c27e323a79ef87c35756ce1e ""
9e07ce365903116b62ac3ac0a033167853853074313f443d5b372f0225eede50 "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
f88600f4b65211a95c6817d0840e0fc2d422883ddf310f29fa8d4cbfda962626 "The quick brown fox jumps over the lazy dog"
ec05208dd1fbf47b9539a761af723612eaa810762ab7a77b715fcfb3bf44f04a "The quick brown fox jumps over the lazy cog"
822578f80d46184a674a6069486b4594053490de8ddf343cc1706418e527bec8 "The quick brown fox jumps over the lazy dog."
410427b981efa6ef884cd1f3d812c880bc7a37abc7450dd62803a4098f28d0f1 "After the rainstorm comes the rainbow."
47b5d8cb1df8d81ed23689936d2edaa7bd5c48f5bc463600a4d7a56342ac80b9 "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
```

Crazy thing is the NodeJS test vectors (from compiled wasm ~~ compiled in Makefile), are all correct except for 1 vector! WTF.

The above is output of:

```
node js/rainsum3.mjs
```

versus

```
rainsum -a storm --test-vectors
```

Why would this even happen! :) 

This is my first time inot wasm

**Comment by crisdosaygo** [comment]
maybe need to: https://developer.chrome.com/docs/devtools/wasm/

**Comment by crisdosaygo** [comment]
Closed by #45
